tasks/clear-cache-reference-data.js

```diff
@@ -0,0 +1,57 @@
+const redisStore = require('../config/redis-store.js')
+
+const scanAndDelete = async (cursor, pattern) => {
+  const client = await redisStore().client
+  return await client
+    .scanAsync(cursor, 'MATCH', pattern, 'COUNT', '100')
+    .then(async ([nextCursor, keys]) => {
+      keys.forEach(key => client.del(key))
+
+      if (nextCursor !== '0') {
+        return keys.length + (await scanAndDelete(nextCursor, pattern))
+      }
+
+      return keys.length
+    })
+}
+
+const clearCacheReferenceData = async referenceName => {
+  if (!referenceName) {
+    return -1
+  }
+
+  return await scanAndDelete(
+    '0',
+    `cache:v2:GET./api/reference/${referenceName}*`
```

Comment:
You might want to incorporate the configuration variables that are used [when building this key](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blob/c1c8c3ae9967b4dd9c17bc94aa8b311bfbb70ba9/common/lib/api-client/middleware/cache-key.js#L18).

If we change the version on the API or the API path I don't think this will work any longer.

Reply:
I've added the version number from the config, but I couldn't find `api/reference` defined anywhere